content/source/docs/enterprise/before-installing/index.html.md

```diff
@@ -137,13 +137,22 @@ For other Linux distributions, check Docker compatibility:
 
 ~> **Important:** We do not recommend running Docker under a 2.x kernel.
 
-### AWS-Specific Configuration
+### IAM Policies
 
-Terraform Enterprise's instance profile serves as default credentials for Terraform's AWS provider. Workspaces without environment variables for credentials will attempt to use the instance profile to provision AWS resources.
+If you have chosen the [external services operational mode](https://www.terraform.io/docs/enterprise/before-installing/index.html#operational-mode-decision), Terraform Enterprise will need access to an S3-compliant endpoint for object storage. You can grant access to the object storage endpoint by either assigning an AWS instance profile or an equivalent IAM system in non-AWS environments.
 
-The instance profile of Terraform Enterprise's instance is the operator's responsibility. If you plan to specify any non-default permissions for Terraform Enterprise's instance profile, be aware that Terraform runs might use those permissions and plan accordingly.
+#### Using the Instance Profile as Default Credentials for Terraform Runs
+You can use Terraform Enterprise's instance profile to provide default credentials to workspaces. Terraform will attempt to use the instance profile to provision resources when you do not set credentials as environment variables. However, this approach presents a few security risks: 
```

Comment:
I know I know, I already edited this! But I reread this and think this might flow better and be cleaner. The point (as I see it) is that the user has two choices, and I want to make them both clear here. This is just me reordering the nice work you already did here!

```suggestion
To set credentials for workspaces, you can: 
- Add credentials as environment variables in each workspace. 
    ~> **Important:** We highly recommend that you [restrict build worker metadata access](../system-overview/security-model.html#restrict-terraform-build-worker-metadata-access) to prevent workspaces from accessing the instance profile.
- Use Terraform Enterprise's instance profile to provide default credentials. Terraform will attempt to use the instance profile to provision resources in all workspaces that do not have credentials as environment variables. This approach presents a few security risks: 
```

Reply:
Hmmm... I agree that this reads better, but now I'm wondering if this really belongs in the "before installing" documentation. TFE users don't really need to think about workspace credentials until after installation.  My goal was to simply add a disclaimer to this existing documentation ("if you do this, be aware of these security implications") rather than provide much instruction.

So I kinda prefer the previous text. What do you think? Any better way to arrange this content with the new disclaimer?

Reply:
ahh okay yes, that's fair. So I have two thoughts. First, maybe we're trying to do too much here (as you pointed out). Maybe all we need to do is be like "hey, think through the permissions you're setting in your instance profile, because they could be used to provision resources down the line." To me, that means we cut this entire section and just have a note under the AWS instance profile sentence. 

 "~>**Important:**  Terraform uses the instance profile as default credentials for workspaces that do not have credentials set as environment variables. That means Terraform could use the permissions in the instance profile to provision resources."

But then - this nice section we made is good information. AND I can't really find it anywhere else (am I missing the page it's on? Is there a good page for us to add this to? I feel like users should know it!

Reply:
Ok, here's my latest (and greatest?) idea on this: I'd like to keep all this content, since I think it is valuable and not stated elsewhere in our docs. I've rearranged the sections though - the heading is "IAM Policies", and the section goes on to explain the required s3 and kms policies, plus the option to use the instance profile to provide default credentials.

How does that sound?

Reply:
okay! Works for me - thank you :-) I just suggest shortening the title to get rid of "using" at the front (see my comment). We're trying to move away from that in general, and this way the title matches the others more closely. Otherwise, feel free to merge today!